modules/core/server/controllers/core.routes.controller.js

```diff
@@ -241,6 +241,30 @@ var setModel = function (Dbclass, name) {
 	};
 };
 exports.setModel = setModel;
+
+var setModelByQuery = function () {
+	return function (req, res, next) {
+		var name = req.query.collection;
+		var Dbclass;
+		var fields;
+		switch(name) {
+			case 'documents':
+				Dbclass = require (path.resolve('./modules/documents/server/controllers/core.document.controller.js'));
```

Comment:
You only have one there, it's not very explicit.  Are you checking that require worked, or that name=documents.  Either way it looks odd.